msal/application.py

```diff
@@ -160,6 +159,39 @@ class ClientApplication(object):
 
     ATTEMPT_REGION_DISCOVERY = True  # "TryAutoDetect"
 
+    _known_authorities = None
+
+    @classmethod
+    def set_known_authorities(cls, known_authorities):
+        """Declare a list of domain names which you allow MSAL to operate with.
+
+        If your app operates with some authorities that you know and own,
+        such as some ADFS or B2C or private cloud,
+        it is recommended and sometimes required that you declare them here,
+        so that MSAL will use them without any validation.
```

Comment:
I do not like the framing here. Can we say: "MSAL will use the endpoints without discovery" instead? "Validation" is what we call the "discovery"

Reply:
Good point about "validation" VS "discovery". I would suggest to avoid the term "endpoint", though, because (1) the endpoint concept is not established in this context; (2) actually we still do a "tenant discovery" (i.e. openid config) to discover the authorization endpoint and token endpoint etc..

So, my next commit will change it to "MSAL will use _these authorities_ without _discovery_".

By the way, are you also OK with those acceptance test cases below, as well as the `known_authority_hosts` naming? Please leave your comment or approval in both this PR and the [internal ApiReview PR](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/6030?_a=files&path=%2FAuthorityValidation%2Fauthority_validation_concepts_and_api.md), then.